code/modules/mod/modules/modules_timeline.dm

```diff
@@ -0,0 +1,448 @@
+	allowed_in_phaseout = TRUE
+	removable = FALSE //copy paste this comment - no timeline modules should be removable
+	///the dummy for phasing from this module, the wearer is phased out while this exists.
+	var/obj/effect/dummy/phased_mob/chrono/phased_mob
+
+/obj/item/mod/module/timeline_jumper/on_use()
+	. = ..()
+	if(!.)
+		return
+	var/area/noteleport_check = get_area(mod.wearer)
+	if(noteleport_check && noteleport_check.area_flags & NOTELEPORT)
+		to_chat(mod.wearer, span_danger("Some dull, universal force is between you and the [phased_mob ? "current timeline" : "stream between timelines"]."))
+		return FALSE
+
+	if(!phased_mob)
+		//phasing out
+		mod.visible_message(span_warning("[mod.wearer] leaps out of the timeline!"))
+		mod.wearer.SetAllImmobility(0)
+		mod.wearer.setStaminaLoss(0, 0)
+		phased_mob = new(get_turf(mod.wearer.loc))
+		mod.wearer.forceMove(phased_mob)
+		RegisterSignal(mod.wearer, COMSIG_MOB_SAY, .proc/handle_speech)
+		RegisterSignal(mod, COMSIG_MOD_ACTIVATE, .proc/on_activate_block)
+	else
+		//phasing in
+		QDEL_NULL(phased_mob)
+		UnregisterSignal(mod, list(COMSIG_MOD_ACTIVATE, COMSIG_MOB_SAY))
+		mod.visible_message(span_warning("[mod.wearer] drops into the timeline!"))
+
+	//probably justifies its own sound but whatever
+	playsound(src, 'sound/items/modsuit/time_anchor_set.ogg', 50, TRUE)
+
+///Signal fired from when wearer speaks while phased out, to relay the speech to the phased dummy
+/obj/item/mod/module/timeline_jumper/proc/handle_speech(datum/source, list/speech_args)
+	SIGNAL_HANDLER
+	//the men are talking
+	phased_mob.say(speech_args[SPEECH_MESSAGE], sanitize = FALSE)
+	//so shut up
+	speech_args[SPEECH_MESSAGE] = ""
+
+///Signal fired when wearer attempts to activate/deactivate suits while phased out
+/obj/item/mod/module/timeline_jumper/proc/on_activate_block(datum/source, user)
+	SIGNAL_HANDLER
+	//has to be a to_chat because you're phased out.
+	to_chat(user, span_boldwarning("Deactivating your suit while inbetween timelines would be a very bad idea."))
+	return MOD_CANCEL_ACTIVATE
+
+///special subtype for phased mobs for the agent to speak from.
+/obj/effect/dummy/phased_mob/chrono
+	name = "reality static"
+	verb_say = "echoes"
+
+///TEM - Lets you eradicate people.
+/obj/item/mod/module/tem
+	name = "MOD timestream eradication module"
+	desc = "The correction device of a fourth dimensional group outside time itself used to \
+			change the destination of a timeline. this device is capable of wiping a being from the \
+			timestream. They never are, they never were, they never will be."
+	icon_state = "chronogun"
+	module_type = MODULE_ACTIVE
+	complexity = 3
+	use_power_cost = DEFAULT_CELL_DRAIN * 5
+	incompatible_modules = list(/obj/item/mod/module/tem)
+	cooldown_time = 0.5 SECONDS
+	removable = FALSE //copy paste this comment - no timeline modules should be removable
+	///reference to the chrono field being controlled by this module
+	var/obj/structure/chrono_field/field = null
```

Comment:
Does this need to be a hard reference rather than a weakref, seems like something that can come and go on a whim?

Reply:
i can confirm, in the case it has a reference it clears the variable, and the case where it doesn't... no issue!